packages/astro/src/runtime/server/render/util.ts

```diff
@@ -196,3 +196,25 @@ export function renderToBufferDestination(bufferRenderFunction: RenderFunction):
 		},
 	};
 }
+
+export const isNode = typeof process !== 'undefined' && Object.prototype.toString.call(process) === '[object process]';
+
+// We can get rid of this when Promise.withResolvers() is ready
+export type PromiseWithResolvers<T> = {
+	promise: Promise<T>
+	resolve: (value: T) => void;
+	reject: (reason?: any) => void;
+}
+
+export function promiseWithResolvers<T = any>(): PromiseWithResolvers<T> {
```

Comment:
Can you add some documentation that explains the utility?

Reply:
Added